docs/building-a-form/creating-a-form-config-file.md

```diff
@@ -1,286 +1,173 @@
-        field1: {
-          'ui:title': 'My field'
-        }
-      }
-    }
-  }
+```
+{
+  type: 'string',
+  enum: ['first', 'second', 'third']
 }
 ```
 
-### About the `schema` and `uiSchema` objects
+The hierarchy for this field uses `SelectWidget` instead of `TextWidget`, because `StringField` detected the `enum` property in the schema.
 
-`uiSchema` is the object that provides information about how the page should be rendered. This object follows the format described in the [react-jsonschema-form documentation](https://github.com/mozilla-services/react-jsonschema-form#the-uischema-object), with some custom us-forms-system additions. The `schema` and `uiSchema` objects should have a similar structure, with the same fields organized in the same way with these exceptions:
+```
+<SchemaField>
+  <StringField>
+    <FieldTemplate>
+      <SelectWidget/>
+    </FieldTemplate>
+  </StringField>
+</SchemaField>
+```
 
-- `uiSchema` doesn't need to contain all the fields found in the `schema` object.
-- `uiSchema` doesn't need a `properties` object for sub-fields.
+While in most cases a field component is responsible for rendering a label and a widget, for `object` and `array` schema types, the field component renders additional field components for each of the elements they contain.
 
-For example, given this schema:
+##### Example schema: `object`
 
-```js
+This is an `object` schema with two string fields.
+
+```
 {
   type: 'object',
   properties: {
     field1: {
       type: 'string'
+    },
+    field2: {
+      type: 'string'
     }
   }
 }
 ```
 
-The matching `uiSchema` would be:
+The `ObjectField` component renders a `SchemaField` component for each of its properties. Those properties are both `string` types, so it looks like the first hierarchy, but nested.
 
-```js
-{
-  'ui:title': 'My form',
-  field1: {
-    'ui:title': 'My field'
-  }
-}
+```
+<SchemaField>
+  <ObjectField>
+    <SchemaField>
+      <StringField>
+        <FieldTemplate>
+          <TextWidget/>
+        </FieldTemplate>
+      </StringField>
+    </SchemaField>
+    <SchemaField>
+      <StringField>
+        <FieldTemplate>
+          <TextWidget/>
+        </FieldTemplate>
+      </StringField>
+    </SchemaField>
+  </ObjectField>
+</SchemaField>
 ```
 
-For array fields, you must specify an `items` object that contains the fields for each row in the array in the `uiSchema` object:
+##### Example schema: `ArrayField`
 
-```js
+`ArrayField` renders a `SchemaField` component for each item in the array. The library only uses the array field where each item is an object type schema.
+
+```
 {
-  'ui:title': 'My form',
-  toursOfDuty: {
-    items: {
-      branchName: {
-        'ui:title': 'Branch'
+  type: 'array',
+  items: {
+    type: 'object',
+    properties: {
+      field1: {
+        type: 'string'
+      },
+      field2: {
+        type: 'string'
       }
     }
   }
 }
 ```
 
-### Configuring `uiSchema` using rjsf options
+### Field components
```

Comment:
I don't think this is a clear title for what this section of content is describing. What it's really describing are the props that Field components pass down through the component hierarchy in order for each component to know what to render. I would instead title it "Field component props".